ocpmodels/models/se3.py

```diff
@@ -0,0 +1,213 @@
+        self.use_pbc = use_pbc
+        self.regress_forces = regress_forces
+
+        self.fibers = {
+            "in": Fiber(1, atom_feature_size),
+            "mid": Fiber(num_degrees, self.num_channels),
+            "out": Fiber(1, num_degrees * self.num_channels),
+        }
+
+        self.embedding = Embedding(100, atom_feature_size)
+
+        blocks = self._build_gcn(self.fibers, 1)
+        self.Gblock, self.FCblock = blocks
+        print(self.Gblock)
+        print(self.FCblock)
+
+    def _build_gcn(self, fibers, out_dim):
+        # Equivariant layers
+        Gblock = []
+        fin = fibers["in"]
+        for i in range(self.num_layers):
+            Gblock.append(
+                GSE3Res(
+                    fin,
+                    fibers["mid"],
+                    edge_dim=self.edge_dim,
+                    div=self.div,
+                    n_heads=self.n_heads,
+                )
+            )
+            Gblock.append(GNormSE3(fibers["mid"]))
+            fin = fibers["mid"]
+        Gblock.append(
+            GConvSE3(
+                fibers["mid"],
+                fibers["out"],
+                self_interaction=True,
+                edge_dim=self.edge_dim,
+            )
+        )
+
+        # Pooling
+        if self.pooling == "avg":
+            Gblock.append(GAvgPooling())
+        elif self.pooling == "max":
+            Gblock.append(GMaxPooling())
+
+        # FC layers
+        FCblock = []
+        FCblock.append(
+            nn.Linear(
+                self.fibers["out"].n_features, self.fibers["out"].n_features
+            )
+        )
+        FCblock.append(nn.ReLU(inplace=True))
+        FCblock.append(nn.Linear(self.fibers["out"].n_features, out_dim))
+
+        return nn.ModuleList(Gblock), nn.ModuleList(FCblock)
+
+    def forward(self, data):
+        pos = data.pos
+        if self.regress_forces:
+            pos = pos.requires_grad_(True)
+
+        if self.use_pbc:
+            out = get_pbc_distances(
+                pos,
+                data.edge_index,
+                data.cell,
+                data.cell_offsets,
+                data.neighbors,
+            )
+
+            data.edge_index = out["edge_index"]
+
+        G = self.pyg2dgl(data)
+
+        # Compute equivariant weight basis from relative positions
+        basis, r = get_basis_and_r(G, self.num_degrees - 1)
+
+        # encoder (equivariant layers)
+        h = {"0": G.ndata["f"]}
+        for layer in self.Gblock:
+            h = layer(h, G=G, r=r, basis=basis)
+
+        for layer in self.FCblock:
+            h = layer(h)
+
+        energy = h
+        if self.regress_forces:
+            forces = -1 * (
+                torch.autograd.grad(
+                    energy,
+                    pos,
+                    grad_outputs=torch.ones_like(h),
+                    create_graph=True,
+                )[0]
+            )
+            return energy, forces
+        else:
+            return energy
```

Comment:
```suggestion

        return h
```
We can just keep this for energy or position?